app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaAppState.kt

```diff
@@ -133,6 +130,17 @@ class NiaAppState(
                 initialValue = emptySet(),
             )
 
+    @OptIn(ExperimentalMaterial3AdaptiveApi::class)
+    val navigationSuiteType: NavigationSuiteType
+        @Composable get() =
+            if (LocalConfiguration.current.screenWidthDp >= 1240) {
```

Comment:
Can we use the new `windowSizeAsState` here instead of querying the `LocalConfiguration`? That is the backing value that drives the `windowSizeClass` calculation, and it can differ from the `LocalConfiguration` value slightly.
Also for testing, it might be better if it was passed into `NiaAppState` similarly to the `windowSizeClass`.

Reply:
Hey Alex, good idea. However

1. `windowSizeAsState` returns screen size in pixels (`IntSize`). However, for the purposes of nav component selection logic you need a `DpSize`. I think it would useful if the `windowSizeAsState` returned `DpSize` directly so there's no need for conversion.
2. I do not see a need for a `windowSizeClass` at all. It's not flexible enough since the `Expanded` case covers both `NavRail` and `PermanentDrawer`. If I have screen size in Dp I don't need `windowSizeClass` anymore.

What do you think?

Reply:
@alexvanyo ping 😄

Reply:
1. Totally agreed
2. The 1240dp breakpoint is the odd one out for not aligning with the window size class, do you think it'd be too verbose to say `windowSizeClass.widthSizeClass == WindowWidthSizeClass.Expanded && windowSize.width >= 1240.dp`?

   Not using it here definitely keeps the code a bit simpler, though could you add a comment to say that the `600.dp` width aligns with the MEDIUM breakpoint?

Reply:
The problem with `WindowSizeClass` visualised:

![window_size_class_vs_m3_vs_camal](https://github.com/android/nowinandroid/assets/6339497/9716ee46-d619-4776-88ea-5905beb52d05)

Reply:
@alexvanyo I added a comment to the navigationSuiteType property explaining the selection logic of the Navigation Component

Reply:
After the internal discussion: how do you feel about removing the customization here entirely, and instead just rely on the logic that is provided by default by the library (which may still be subject to change)

Reply:
@alexvanyo To better accommodate large-screen devices, we strongly advocate for the inclusion of a Permanent Drawer.